make_a_plea/management/commands/zendesk.py

```diff
@@ -0,0 +1,234 @@
+  #- "?query=type:ticket -tags:plea_receipt -description:\"Used the call centre\" -description:\"Comments -\""
+
+Usage
+=====
+
+./manage.py zendesk --to-csv
+
+bugs and limitations
+====================
+
+   * Where 'received_at:makeaplea@ministryofjustice.zendesk.com' is used in
+   the web query builder, it is not clear from which field this derives, nor
+   does it appear possible to search on subkeys on the 'via' field. Also,
+   'recipient' is not the field as there are too few results. We could post-
+   process to fix this, currently the results ignore this filter.
+   * There is no de-duplication between filters
+   * There is no separation of filters into different files
+   * zdesk failed (subjects were often truncated) when used in normal mode.
+   Since raw_queries are now required and the caching is of little use to us
+   for this task we could switch out to just requests.
+
+"""
+
+import codecs
+import csv
+import cStringIO
+import os
+from optparse import make_option
+import re
+import yaml
+
+
+from django.core.management.base import BaseCommand
+from zdesk import Zendesk
+
+
+# Most fields are simple scalars. Fields should only be in one list.
+SIMPLE_FIELDS = [
+    u'id',
+    u'status',
+    u'created_at',
+    u'updated_at',
+    u'due_at',
+    u'group_id',
+    u'has_incidents',
+    u'is_public',
+    u'organization_id',
+    u'priority',
+    u'problem_id',
+    u'recipient',
+    u'requester_id',
+    u'satisfaction_rating',
+    u'status',
+    u'submitter_id',
+    u'type',
+    u'url',
+    u'result_type',
+    u'allow_channelback',
+    u'assignee_id',
+    u'brand_id',
+]
+
+# Some have structure like dicts or lists that will be strigified
+STRUCTURED_FIELDS = [
+    u'tags',
+    u'external_id',
+    u'fields',
+    u'forum_topic_id',
+    u'followup_ids',
+    u'collaborator_ids',
+    u'custom_fields',
+    u'sharing_agreement_ids',
+    u'via',
+]
+
+# Some may contain various line endings that will be coerced to \n
+NEWLINE_FIELDS = [
+    "description",
+    "subject",
+    "raw_subject",
+]
+
+
+class Command(BaseCommand):
+
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '--to-csv',
+            action='store_true',
+            dest='to-csv',
+            default=False,
+            help='Export all relevant zendesk items to csv file'),
+        )
+
+    def handle(self, *args, **options):
+        self.load_settings()
+        self.init_zen()
+
+        if options["to-csv"]:
+            self.to_csv()
+
+    def load_settings(self):
+        with open(os.path.expanduser("~/.hmcts-zendesk"), "r") as config_file:
+            self.settings = yaml.load(config_file.read())
+
+    def init_csv(self, f, dialect=csv.excel, encoding="utf-8", **kwds):
+        self.queue = cStringIO.StringIO()
+        self.writer = csv.writer(self.queue, dialect=dialect, **kwds)
```

Comment:
I found myself at https://docs.python.org/2/library/csv.html#writer-objects and overlooked the simpler example. Or it didn't work at the time. Ripped out much pointless encoding cruft, thanks.